code/SubsitesVirtualPage.php

```diff
@@ -230,4 +230,12 @@ public function init()
         
         Subsite::$disable_subsite_filter = $origDisableSubsiteFilter;
     }
+    
+    public function getViewer($action) {
+        $originalClass = get_class($this->CopyContentFrom());
+        if ($originalClass == 'SiteTree') $name = 'Page_Controller';
+        else $name = $originalClass."_Controller";
+        $controller = new $name($this->CopyContentFrom());
```

Comment:
This isn't the correct way to get a controller for a model.

use `ModelAsController`

```php
$controller = ModelAsController::controller_for($this->CopyContentFrom());
$controller->init();
return $controller->getViewer($action);
```